actors/builtin/miner/deadline_state.go

```diff
@@ -444,3 +446,161 @@ func (dl *Deadline) popExpiredPartitions(store adt.Store, until abi.ChainEpoch,
 
 	return popped, modified, nil
 }
+
+// RemovePartitions removes the specified partitions, shifting the remaining
+// ones to the left, and returning the live and dead sectors they contained.
+//
+// Returns an error if any of the partitions contained faulty sectors or early
+// terminations.
+func (dl *Deadline) RemovePartitions(store adt.Store, partitions *bitfield.BitField, quant QuantSpec) (
+	live, dead *abi.BitField, removedPower PowerPair, err error,
+) {
+	oldPartitions, err := dl.PartitionsArray(store)
+	if err != nil {
+		return nil, nil, NewPowerPairZero(), xerrors.Errorf("failed to load partitions: %w", err)
+	}
+
+	partitionCount := oldPartitions.Length()
+	toCompact, err := partitions.AllMap(partitionCount)
```

Comment:
`toCompact` and all references to compaction are confusing - this method only does the removal part and is quite well defined as doing just that.